docs: update the version specific notes table

bmcustodio opened the pull request:
Updates the table in the "Version Specific Notes" subsection of the "Upgrade" page in order to be explicit about the supported upgrade paths.

qmonnet commented on Documentation/operations/upgrade.rst:
```diff
@@ -283,13 +283,15 @@ transitions, ordered by version.
-case, consider staging the upgrade, for example upgrading from ``1.1.x`` to the
-latest ``1.1.y`` release before subsequently upgrading to ``1.2.z``.
+case, consider performing incremental upgrades between versions (e.g. upgrade
+from ``1.9.x`` to ``1.10.y`` first, and only after that upgrade to ``1.11.y``).
```
Slightly confusing how it reads that the second step should be upgrading from `1.10.y` to `1.11.y`, and this `y` seems to be the same minor version number for both versions. I'd rather keep a `z` personally, but no strong opinion.

bmcustodio committed:
docs: update the version specific notes table

Updates the table in the "Version Specific Notes" subsection of the
"Upgrade" page in order to be explicit about the supported upgrade
paths.

Signed-off-by: Bruno Miguel Custódio <brunomcustodio@gmail.com>
